Docs/HowFuzzilliWorks.md

```diff
@@ -766,29 +766,29 @@ As previously discussed, there needs to be some guidance mechanism to direct the
 An example for a ProgramTemplate is shown next.
 
 ```swift
-ProgramTemplate("JITFunction") { b in
+fileprivate let JITFunction = ProgramTemplate("JITFunction") { b in
     // Generate some random header code
     b.generate(n: 10)
 
-    let sig = b.randFunctionSignature() // invokes TypeGenerators
-    let f = b.defineFunction(withSignature: sig) { b, params in
-        b.generate(n: 100)
+    var signature = ProgramTemplate.generateRandomFunctionSignatures(forFuzzer: b.fuzzer, n: 1)[0] // invokes TypeGenerators
+
+    // Here generate a random function type, e.g. arrow/generator etc
+    let f = b.definePlainFunction(withSignature: signature) { args in
+        b.generate(n: 10)
     }
+    
 
     // Generate some random code in between
     b.generate(n: 10)
-
-    // Call the function repeatedly to trigger JIT compilation
-    b.forLoop(from: 0, to: 100) { _ in
-        // Find or instantiate the required argument values
-        let args = b.generateArguments(for: f)
-        b.callFunction(f, withArgs: args)
+    b.forLoop(b.loadInt(0), .lessThan, b.loadInt(20), .Add, b.loadInt(1)) { args in
+        b.callFunction(f, withArgs: b.generateCallArguments(for: signature))
     }
-
+    
     // Call the function again with different arguments
-    let args = b.generateArguments(for: f)
-    b.callFunction(f, withArgs: args)
-}
+    let args = b.randCallArguments(for: f)
```

Comment:
I think this can be `generateCallArguments` and then doesn't need to be unwrapped: https://github.com/googleprojectzero/fuzzilli/blob/main/Sources/Fuzzilli/Core/ProgramTemplates.swift#L67

Reply:
got it